stl/inc/numbers

```diff
@@ -0,0 +1,188 @@
+// numbers standard header (core)
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _NUMBERS_
+#define _NUMBERS_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if !_HAS_CXX20
+#pragma message("The contents of <numbers> are available only with C++20 or later.")
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+#ifdef __cpp_lib_concepts
+#include <concepts>
+#endif
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+namespace numbers {
+    template <class _Ty>
+    struct _Invalid {
+        static_assert(_Always_false<_Ty>, "This variable template is not specialised for this type which is invalid.");
+    };
+    template <class _Ty>
+    inline constexpr _Ty e_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty log2e_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty log10e_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty pi_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty inv_pi_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty inv_sqrtpi_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty ln2_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty ln10_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty sqrt2_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty sqrt3_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty inv_sqrt3_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty egamma_v = _Invalid<_Ty>();
+    template <class _Ty>
+    inline constexpr _Ty phi_v = _Invalid<_Ty>();
+#ifdef __cpp_lib_concepts
+    template <floating_point _Floating>
+    inline constexpr _Floating e_v<_Floating> = static_cast<_Floating>(2.718281828459045);

```

Comment:
Should I just use auto return type for the specialisations?

Reply:
Your call. We often say `auto` when there's a cast on the RHS, since that provides clear type information (especially if the type is compound like `const char*`). Using it here would be consistent with that principle. But because `_Floating` is simple and a short name, I view `auto` as providing less benefit. I would be happy either way.